### PR TITLE
Transaction external type

### PR DESCRIPTION
**Pull request by ashkan18**
# Change
This adds `external_type` to `transactions` table, this way we can distinguish between `charge`, `refund` and newly added `payment_intent` transactions.
This way when trying to `capture` or `refund` we first find the `transaction` for this `external_id` and based on it's type decide if we need to do a `charge` or `payment_intent` refund/capture.

### CODE REVIEW

File: app/graphql/mutations/order_or_failure_union_type.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class Mutations::OrderWithMutationFailure < Types::BaseObject
   field :error, Types::ApplicationErrorType, null: false
 end
 
-class Mutations::OrderNeedsAction < Types::BaseObject
+class Mutations::OrderRequiresAction < Types::BaseObject
```
Comment (sweir27) [inline app/graphql/mutations/order_or_failure_union_type.rb]: Why this change?

Comment (sweir27) [reply app/graphql/mutations/order_or_failure_union_type.rb]: Also does the `OrderOrFailureUnionType` need to have this as part of the `possible_types`?

Comment (ashkan18) [reply app/graphql/mutations/order_or_failure_union_type.rb]: this new name matches the actual `transaction_status`. Yes, we added it to union type.

File: db/migrate/20190717171856_populate_transaction_external_type.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class PopulateTransactionExternalType < ActiveRecord::Migration[5.2]
+  def up
+    Transaction.where('external_id LIKE :prefix', prefix: 'ch_').update_all(external_type: Transaction::CHARGE)
+    Transaction.where('external_id LIKE :prefix', prefix: 're_').update_all(external_type: Transaction::REFUND)
```
Comment (ashkan18) [inline db/migrate/20190717171856_populate_transaction_external_type.rb]: we also have `re_` transactions which are for `reufnds` so we need to update them, the tricky thing about refunds is their `transaction_type` and `external_type` is both `Transaction::REFUNDS`

File: lib/payment_service.rb [hunk 3]
```diff
@@ -1,44 +1,59 @@
...
+  def self.hold_charge(charge_params)
     create_payment_intent(charge_params.merge(capture: false))
   end
 
-  def self.create_and_capture_charge(charge_params)
+  def self.immediate_capture(charge_params)
     create_payment_intent(charge_params.merge(capture: true))
   end
 
   def self.capture_authorized_charge(external_id)
```
Comment (ashkan18) [inline lib/payment_service.rb]: im not in ❤️ with this method getting `external_id` and not our internal `transaction` `id`. but this may be a bigger change.
